integr8ly/delorean
# Adds extract-bundle and process-bundle commands

# Issue link
https://issues.redhat.com/browse/MGDAPI-2465

# What
With the onboarding of RHOAM to the Managed Service Pipeline, product bundles need to be mirrored and included in the RHOAM index. By adding two commands (`extract-bundle` and `process-bundle`) to the current EWS framework, our existing Jenkins jobs can be modified to submit a PR that updates the bundle reference for a product whenever a new version of that product comes across the UMB.

# Verification steps
- Prerequisite: Checkout the master branch of [integreatly-operator](https://github.com/integr8ly/integreatly-operator) to your local machine and note the location for later
- Prerequisite: Create a directory to store the extracted bundle, for example: `mkdir ~/Documents/3scale-bundle-extracted`
- Checkout this PR and enter the delorean directory
- Build the delorean CLI: `make build/cli`
- Test the `extract-bundle` command using a 3scale bundle: 
```
./delorean ews extract-bundle --extract-dir ~/Documents/3scale-bundle-extracted --src-image=registry-proxy.engineering.redhat.com/rh-osbs/3scale-amp2-3scale-rhel7-operator-metadata@sha256:6c916f91899e1c280859ccc49aea9c16554d92e33857151b1050bffc905cb872 
```
- Test the `process-bundle` command using the extracted 3scale bundle (change the `--products-path` to wherever your copy of integreatly-operator is located):
```
./delorean ews process-bundle --product 3scale --bundle-dir ~/Documents/3scale-bundle-extracted --products-path ~/Documents/integreatly-operator/products/installation-cpaas.yaml --bundle this-is-a-test-bundle
```
- Go to your local copy of `integreatly-operator/products/installation-cpaas.yaml` and confirm that it was updated with the new 3scale information.
